app/tournament/tournament.js

```diff
@@ -0,0 +1,93 @@
+import {Player, validatePlayer} from "./player";
+import {Match, ChildMatch} from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        this.players.push(new Player(player.name, player.rank));
+        sortPlayersByRank(this);
+    }
+
+    addMatch(match) {
+        this.matches.push(match);
+    }
+
+    getRounds() {
```

Comment:
I don't know what to expect from this method. A number? A list of rounds? Even reading the implementation, I don't know. Use unambiguous names.